Adopt GB18030-2022 <-> Unicode mapping recommendation

vitorroriz opened the pull request:
#### 4659e88a8afd048b77069f7aca297ef4b50b1200
<pre>
Adopt GB18030-2022 &lt;-&gt; Unicode mapping recommendation
<a href="https://bugs.webkit.org/show_bug.cgi?id=279761">https://bugs.webkit.org/show_bug.cgi?id=279761</a>
<a href="https://rdar.apple.com/124367525">rdar://124367525</a>

Reviewed by Alex Christensen.

As per [1] the Unicode Technical Committee recommends an updated mapping between GB18030 and Unicode
which we follow here.

Related tests need to be updated and exported to WPT. The PDF [1] contains a text
file attachment with the recommended mapping, which is:

Transcoding From Unicode Standard to GB 18030:
U+E78D -&gt; 0xA6D9
U+E78E -&gt; 0xA6DA
U+E78F -&gt; 0xA6DB
U+E790 -&gt; 0xA6DC
U+E791 -&gt; 0xA6DD
U+E792 -&gt; 0xA6DE
U+E793 -&gt; 0xA6DF
U+E794 -&gt; 0xA6EC
U+E795 -&gt; 0xA6ED
U+E796 -&gt; 0xA6F3
U+E81E -&gt; 0xFE59
U+E826 -&gt; 0xFE61
U+E82B -&gt; 0xFE66
U+E82C -&gt; 0xFE67
U+E832 -&gt; 0xFE6D
U+E843 -&gt; 0xFE7E
U+E854 -&gt; 0xFE90
U+E864 -&gt; 0xFEA0

Transcoding From GB 18030 to Unicode Standard:
0x82359037 -&gt; U+9FB4
0x82359038 -&gt; U+9FB5
0x82359039 -&gt; U+9FB6
0x82359130 -&gt; U+9FB7
0x82359131 -&gt; U+9FB8
0x82359132 -&gt; U+9FB9
0x82359133 -&gt; U+9FBA
0x82359134 -&gt; U+9FBB
0x84318236 -&gt; U+FE10
0x84318237 -&gt; U+FE11
0x84318238 -&gt; U+FE12
0x84318239 -&gt; U+FE13
0x84318330 -&gt; U+FE14
0x84318331 -&gt; U+FE15
0x84318332 -&gt; U+FE16
0x84318333 -&gt; U+FE17
0x84318334 -&gt; U+FE18
0x84318335 -&gt; U+FE19

[1] <a href="https://www.unicode.org/L2/L2023/23003r-gb18030-recommendations.pdf">https://www.unicode.org/L2/L2023/23003r-gb18030-recommendations.pdf</a> brings new mapping recommendations between GB18030 and Unicode.

* LayoutTests/imported/w3c/web-platform-tests/encoding/legacy-mb-schinese/gb18030/gb18030-decoder.any.js:
* LayoutTests/imported/w3c/web-platform-tests/encoding/legacy-mb-schinese/gb18030/gb18030-encoder.html:
* Source/WebCore/PAL/pal/text/TextCodecCJK.cpp:
(PAL::gb180302022Encode):
(PAL::gb180302022Decode):

Canonical link: <a href="https://commits.webkit.org/283724@main">https://commits.webkit.org/283724@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bb9b1899628786fea1c8516df1726412cd588dec

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67157 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46536 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19789 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/71192 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/18290 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/69275 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54334 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/18083 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/53849 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12308 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70224 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42800 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/58110 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34384 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39472 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15504 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16644 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/61438 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15845 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72895 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/11116 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15200 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/61323 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/11149 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58169 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61401 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14879 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9137 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2737 "Passed tests") | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/42341 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43418 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44604 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43159 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 reviewed (changes requested):
We do not want to make this change.  These tests have not been upstreamed yet to my knowledge.  This issue has been reported in https://github.com/whatwg/encoding/issues/312

achristensen07 reviewed (approved):
Apparently something had changed that I was not aware of until today.  Approved.

vitorroriz committed:
Adopt GB18030-2022 <-> Unicode mapping recommendation
https://bugs.webkit.org/show_bug.cgi?id=279761
rdar://124367525

Reviewed by Alex Christensen.

As per [1] the Unicode Technical Committee recommends an updated mapping between GB18030 and Unicode
which we follow here.

Related tests need to be updated and exported to WPT. The PDF [1] contains a text
file attachment with the recommended mapping, which is:

Transcoding From Unicode Standard to GB 18030:
U+E78D -> 0xA6D9
U+E78E -> 0xA6DA
U+E78F -> 0xA6DB
U+E790 -> 0xA6DC
U+E791 -> 0xA6DD
U+E792 -> 0xA6DE
U+E793 -> 0xA6DF
U+E794 -> 0xA6EC
U+E795 -> 0xA6ED
U+E796 -> 0xA6F3
U+E81E -> 0xFE59
U+E826 -> 0xFE61
U+E82B -> 0xFE66
U+E82C -> 0xFE67
U+E832 -> 0xFE6D
U+E843 -> 0xFE7E
U+E854 -> 0xFE90
U+E864 -> 0xFEA0

Transcoding From GB 18030 to Unicode Standard:
0x82359037 -> U+9FB4
0x82359038 -> U+9FB5
0x82359039 -> U+9FB6
0x82359130 -> U+9FB7
0x82359131 -> U+9FB8
0x82359132 -> U+9FB9
0x82359133 -> U+9FBA
0x82359134 -> U+9FBB
0x84318236 -> U+FE10
0x84318237 -> U+FE11
0x84318238 -> U+FE12
0x84318239 -> U+FE13
0x84318330 -> U+FE14
0x84318331 -> U+FE15
0x84318332 -> U+FE16
0x84318333 -> U+FE17
0x84318334 -> U+FE18
0x84318335 -> U+FE19

[1] https://www.unicode.org/L2/L2023/23003r-gb18030-recommendations.pdf brings new mapping recommendations between GB18030 and Unicode.

* LayoutTests/imported/w3c/web-platform-tests/encoding/legacy-mb-schinese/gb18030/gb18030-decoder.any.js:
* LayoutTests/imported/w3c/web-platform-tests/encoding/legacy-mb-schinese/gb18030/gb18030-encoder.html:
* Source/WebCore/PAL/pal/text/TextCodecCJK.cpp:
(PAL::gb180302022Encode):
(PAL::gb180302022Decode):

Canonical link: https://commits.webkit.org/283724@main
